### PR TITLE
Add `NcSavingIndicatorIcon`

### PR DESCRIPTION
**Pull request by susnux**
An icon to indicate the saving state of a document, file... as used by the *text* and the *office* app.
The icon is based on the material design icon `Check` combined with `CircleMedium`.

I think as this is now used by two apps we should provide this for other apps as well so they do not need to reinvent the wheel.

For design review see this PR which introduces the icon for the text app: https://github.com/nextcloud/text/pull/3572

### CODE REVIEW

File: src/components/NcSavingIndicatorIcon/NcSavingIndicatorIcon.vue [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  -->
+
+<docs>
+# Usage
+
+```
+<template>
+<div>
+	<div>
+		<NcSavingIndicatorIcon :title="`${savingState === 'error' ? 'Error saving' : savingState} file`"
```
Comment (nickvergessen) [inline src/components/NcSavingIndicatorIcon/NcSavingIndicatorIcon.vue:28]: This should be translated

Comment (susnux) [reply src/components/NcSavingIndicatorIcon/NcSavingIndicatorIcon.vue:28]: This is just the documentation

Comment (nickvergessen) [reply src/components/NcSavingIndicatorIcon/NcSavingIndicatorIcon.vue:28]: Oh, sorry

**Review by skjnldsv — APPROVED** [review]
Approved by the design team, good from me too  :+1:

**Review by raimund-schluessler — APPROVED** [review]

**Pull request merged**